### PR TITLE
Add an end-point to register new consumption of a feature

### PR DESCRIPTION
**Pull request by eprudnikov**
This draft shows a uniform interface to register the consumption of a feature. It's the uniform one because it encapsulates
- Update of the usage value if necessary
- Check whether the limit (if any) is exceeded
- Withdraw money from the balance if necessary

As a result, clients do not care about any limitation or price model the feature has.
The PR still lacks 
- Request model that will include at least quantity.
- Response model. Ideally, it should contain information about the current usage level and limit if they are set.

### CODE REVIEW

File: hub/hub-api.yaml [hunk 1]
```diff
@@ -1149,6 +1149,50 @@ paths:
             application/json:
               schema:
                 $ref: "#/components/schemas/ResponseError"
+ 
+  /features/{anchor}/usage:
```
Comment (aehrenthal) [inline hub/hub-api.yaml:1153]: Would it be better to use usage or usages?
@Zemidar and @BullDog57Rus what do you think too?

File: hub/hub-api.yaml [hunk 2]
```diff
@@ -1149,6 +1149,50 @@ paths:
...
+    post:
+      summary: Register a usage of a feature.
+      description: |
+        Use this endpoint to inform Hub about the consumption of the feature.
```
Comment (aehrenthal) [inline hub/hub-api.yaml:1157]:
```suggestion
        Use this endpoint to inform the Hub about the consumption of a feature. 
```

File: hub/hub-api.yaml [hunk 3]
```diff
@@ -1149,6 +1149,50 @@ paths:
...
+      description: |
+        Use this endpoint to inform Hub about the consumption of the feature. 
+        As a result, Hub can restrict the use of the feature or withdraw money from the tenant's balance. 
+      operationId: createFeatureUsage
+      tags:
+        - Features
+      parameters:
+        - name: anchor
+          in: path
+          description: A unique anchor of feature.
```
Comment (aehrenthal) [inline hub/hub-api.yaml:1165]:
```suggestion
          description: A unique anchor of the feature.
```

**Review by aehrenthal — CHANGES_REQUESTED** [review]
@eprudnikov I like the idea and concept behind this, very nice work. I assume we will still add some request body or additional parameters to attach additional information such as the quantity of the usage, etc.?

File: hub/hub-api.yaml (hunk 1 above)
Comment (Zemidar) [reply hub/hub-api.yaml:1153]: As for me usages more suitable

Comment (BullDog57Rus) [reply hub/hub-api.yaml:1153]: I think usages makes more sense. And actually maybe it is better to add tenantId as a parameter somehow? Of course we always have the token of the tenant or we can override it using the API KEY. Just an idea

**Comment by eprudnikov** [comment]
Thank you all for the feedback!
The main goal of this draft was to show an idea of the interface. I close it to rework [it based on the up-to-date API version](https://github.com/inperium-corp/inperium-openapi/pull/160). All your suggestions are included there.

**Pull request closed without merging**